component/portal/src/main/java/org/exoplatform/portal/mop/dao/SiteDAOImpl.java

```diff
@@ -123,6 +141,20 @@ public List<String> findUserSites(int offset, int limit) {
     return getSiteNames(SiteType.USER, offset, limit);
   }
 
+  @Override
+  public List<SiteKey> findSitesKeys(SiteFilter siteFilter) {
+    TypedQuery<SiteEntity> query = buildQueryFromFilter(siteFilter);
```

Comment:
```suggestion
    TypedQuery<Tuple> query = buildQueryFromFilter(siteFilter);
```
It would be better to use Tuples or simply instantiate the SiteKey in JPQL to improve the performances of query and to not having to retrieve all the SiteEntity data. ([See it here](https://github.com/Meeds-io/gamification/blob/develop/services/src/main/java/io/meeds/gamification/entity/RealizationEntity.java#L46))

This is not mandatory as improvements since we will not have a lot of Sites to manage at the same time in the table, but just to inform about the best practice: **don't ask DB for more than what you need**

Reply:
fixe with TypedQuery<SiteKey>